src/comps/main/comments/comments.jsx

```diff
@@ -1,37 +1,50 @@
-import React, { useState, useEffect } from "react";
-import comments from "./comment.css";
-import data from "../../../data/data.json";
+import React, { Component } from "react";
 import { allComments } from "../../../data/UserFunctions";
-export const Comments = props => {
-  const [commentList, setCommentList] = useState([]);
-
-  const activeId = props.activeVideoId;
-
-  function getComments() {
-    console.log("in");
+class CommentListComponent extends Component {
+  _isMounted = false;
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeVideo: this.props.activeId,
+      commentList: []
+    };
+  }
+  componentDidMount() {
+    this._isMounted = true;
     allComments()
-      .then(res => setCommentList(res.data))
-      .catch(err => console.log(err));
+      .then(res => this.setState({ commentList: res.data }))
+      .catch(err => err);
   }
 
-  useEffect(() => {
-    getComments();
-  }, []);
-  const filter = commentList.filter(num => num.video_id == activeId);
-  // console.log(filter);
-  return (
-    <React.Fragment>
-      {filter.map((_data, index) => {
-        return (
-          <div className="card w-50 mt-3" key={_data.id}>
+  componentDidUpdate(prevProps) {
```

Comment:
I don't understand why you put this data in state when you can just use it from props

Reply:
the props is just for checking if the video id is valid or was changed 
if it was changed then the component update hook will change the state via the new prevProps video id